"use strict";

const UseCaseError = require("./use-case-error.js");

class UnexpectedError extends UseCaseError {
  constructor(msg, cause = null) {
    super(msg, cause);
    this.status = 500;
    this.message = "Internal server error has occurred.";
    this.code = `${UseCaseError.ERROR_PREFIX}server/internalServerError`;
  }
}

module.exports = UnexpectedError;
